### PR TITLE
Python: enforce pytype in railjson_generator

### LINKED ISSUE #5520: Python: enforce pytype in railjson_generator
### Description and goal

python/ has types, but they're not checked

Next: update the website, and other repos (dags?)

See also: https://github.com/osrd-project/osrd/issues/5508

### Acceptance criteria

Complete when pytype doesn't return errors in railjson_generator and is checked on pull requests.

### PR DESCRIPTION
**Pull request by shenriotpro**
closes https://github.com/osrd-project/osrd/issues/5520

### CODE REVIEW

File: python/railjson_generator/railjson_generator/schema/infra/neutral_section.py [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 
 def _neutral_section_id():
-    res = f"neutral_section.{NeutralSection._INDEX}"
-    NeutralSection._INDEX += 1
+    res = f"neutral_section.{NeutralSection._INDEX}"  # pytype: disable=name-error
+    NeutralSection._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/neutral_section.py]:
```suggestion
    # pytype: disable=name-error
    res = f"neutral_section.{NeutralSection._INDEX}"  
    NeutralSection._INDEX += 1 
    # pytype: enable=attribute-error 
```

File: python/railjson_generator/railjson_generator/schema/infra/signal.py [hunk 2]
```diff
@@ -7,8 +7,8 @@
 
 
 def _signal_id():
-    res = f"signal.{Signal._INDEX}"
-    Signal._INDEX += 1
+    res = f"signal.{Signal._INDEX}"  # pytype: disable=name-error
+    Signal._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/signal.py]:
```suggestion
    # pytype: disable=attribute-error
    res = f"signal.{Signal._INDEX}"  
    Signal._INDEX += 1 
    # pytype: enable=attribute-error
```

File: python/railjson_generator/railjson_generator/schema/infra/speed_section.py [hunk 3]
```diff
@@ -9,8 +9,8 @@
 
 
 def _speed_section_id():
-    res = f"speed_section.{SpeedSection._INDEX}"
-    SpeedSection._INDEX += 1
+    res = f"speed_section.{SpeedSection._INDEX}"  # pytype: disable=name-error
+    SpeedSection._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/speed_section.py]:
```suggestion
    # pytype: disable=name-error
    res = f"speed_section.{SpeedSection._INDEX}"  
    SpeedSection._INDEX += 1
    # pytype: enable=attribute-error
```

File: python/railjson_generator/railjson_generator/schema/infra/switch.py [hunk 4]
```diff
@@ -14,8 +15,8 @@
 
 
 def _switch_id():
-    res = f"switch.{Switch._INDEX}"
-    Switch._INDEX += 1
+    res = f"switch.{Switch._INDEX}"  # pytype: disable=name-error
+    Switch._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/switch.py]: same change as above

File: python/railjson_generator/railjson_generator/schema/infra/track_section.py [hunk 5]
```diff
@@ -15,8 +15,8 @@
 
 
 def _track_id():
-    res = f"track.{TrackSection._INDEX}"
-    TrackSection._INDEX += 1
+    res = f"track.{TrackSection._INDEX}"  # pytype: disable=name-error
+    TrackSection._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/track_section.py]: same change as above

File: python/railjson_generator/railjson_generator/schema/infra/waypoint.py [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
 
-Waypoint = Union[Type["BufferStop"], Type["Detector"]]
+Waypoint = Union["BufferStop", "Detector"]
 
 
 def _buffer_stop_id():
-    res = f"buffer_stop.{BufferStop._INDEX}"
-    BufferStop._INDEX += 1
+    res = f"buffer_stop.{BufferStop._INDEX}"  # pytype: disable=name-error
+    BufferStop._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/waypoint.py]: same change as above

File: python/railjson_generator/railjson_generator/schema/infra/waypoint.py [hunk 7]
```diff
@@ -44,8 +42,8 @@ def get_direction(self, track) -> Direction:
 
 
 def _detector_id():
-    res = f"detector.{Detector._INDEX}"
-    Detector._INDEX += 1
+    res = f"detector.{Detector._INDEX}"  # pytype: disable=name-error
+    Detector._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/infra/waypoint.py]: same here

File: python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py [hunk 8]
```diff
@@ -16,8 +16,8 @@
 
 
 def _train_id():
-    res = f"train.{TrainSchedule._INDEX}"
-    TrainSchedule._INDEX += 1
+    res = f"train.{TrainSchedule._INDEX}"  # pytype: disable=name-error
+    TrainSchedule._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py]: Same here

File: python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py [hunk 9]
```diff
@@ -76,8 +76,8 @@ def format(self):
 
 
 def _group_id():
-    res = f"group.{TrainScheduleGroup._INDEX}"
-    TrainScheduleGroup._INDEX += 1
+    res = f"group.{TrainScheduleGroup._INDEX}"  # pytype: disable=name-error
+    TrainScheduleGroup._INDEX += 1  # pytype: disable=name-error
```
Comment (younesschrifi) [inline python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py]: Same here

**Review by younesschrifi — COMMENTED** [review]
Nice PR, LGTM. 
I've made few comments 😄

**Review by younesschrifi — APPROVED** [review]

**Review by multun — APPROVED** [review]

**Review by leovalais — APPROVED** [review]

**Pull request merged**